vhost-device-spi/src/spi.rs

```diff
@@ -0,0 +1,1055 @@
+                    && (req.tx_nbits != 8))
+            {
+                trans_params_valid = false;
+            }
+
+            if ((self.config.rx_nbits_supported & ConfigNbits::DUAL.bits()) == 0
+                && req.rx_nbits == 2)
+                || ((self.config.rx_nbits_supported & ConfigNbits::QUAD.bits()) == 0
+                    && req.rx_nbits == 4)
+                || ((self.config.rx_nbits_supported & ConfigNbits::OCTAL.bits()) == 0
+                    && req.rx_nbits == 8)
+                || ((req.rx_nbits != 0)
+                    && (req.rx_nbits != 1)
+                    && (req.rx_nbits != 2)
+                    && (req.rx_nbits != 4)
+                    && (req.rx_nbits != 8))
+            {
+                trans_params_valid = false;
+            }
+
+            if (req.mode & ReqMode::CPHA.bits() == 0)
+                && (self.config.mode_func_supported.to_native() & ConfigMode::CPHA_0.bits() == 0)
+            {
+                trans_params_valid = false;
+            }
+
+            if (req.mode & ReqMode::CPHA.bits() == ReqMode::CPHA.bits())
+                && (self.config.mode_func_supported.to_native() & ConfigMode::CPHA_1.bits() == 0)
+            {
+                trans_params_valid = false;
+            }
+
+            if (req.mode & ReqMode::CPOL.bits() == 0)
+                && (self.config.mode_func_supported.to_native() & ConfigMode::CPOL_0.bits() == 0)
+            {
+                trans_params_valid = false;
+            }
+
+            if (req.mode & ReqMode::CPOL.bits() == ReqMode::CPOL.bits())
+                && (self.config.mode_func_supported.to_native() & ConfigMode::CPOL_1.bits() == 0)
+            {
+                trans_params_valid = false;
+            }
+
+            if (req.mode & ReqMode::CS_HIGH.bits() == ReqMode::CS_HIGH.bits())
+                && (self.config.mode_func_supported.to_native() & ConfigMode::CS_HIGH.bits() == 0)
+            {
+                trans_params_valid = false;
+            }
+
+            if (req.mode & ReqMode::LSB_FIRST.bits() == ReqMode::LSB_FIRST.bits())
+                && (self.config.mode_func_supported.to_native() & ConfigMode::LSB.bits() == 0)
+            {
+                trans_params_valid = false;
+            }
+
+            if (req.mode & ReqMode::LOOP.bits() == ReqMode::LOOP.bits())
+                && (self.config.mode_func_supported.to_native() & ConfigMode::LOOP.bits() == 0)
+            {
+                trans_params_valid = false;
+            }
+
+            if (self.config.max_freq_hz != 0)
+                && (req.speed_hz > self.config.max_freq_hz.to_native())
+            {
+                trans_params_valid = false;
+            }
+        }
+
+        trans_params_valid
+    }
+}
+
+#[cfg(test)]
+pub(crate) mod tests {
+    use super::*;
+    use vmm_sys_util::tempfile::TempFile;
+
+    // Update read-buffer of each write-buffer with index + 1 value.
+    pub fn update_rdwr_buf(buf: u64, len: u32) {
```

Comment:
```suggestion
    /// Update read-buffer of each write-buffer with index + 1 value.
    fn update_rdwr_buf(buf: u64, len: u32) {
```

Reply:
Actually `update_rdwr_buf` is an auxiliary function in unit test, do I need to use `///` to comment it?

Reply:
Ah, then no need indeed :)